scripts/deploymentextensions/installtools/install.sh

```diff
@@ -0,0 +1,210 @@
+oxa_tools_public_github_projectname="oxa-tools"
+oxa_tools_public_github_projectbranch="oxa/master.fic"
+oxa_tools_public_github_branchtag=""
+oxa_tools_repository_path="/oxa/oxa-tools"
+
+# Email Notifications
+notification_email_subject="Tools Installer"
+cluster_admin_email=""
+
+# this is the user account that will be used for ssh
+target_user=""
+
+# operation mode: 0=local, 1=remote via ssh
+remote_mode=0
+
+# debug mode: 0=set +x, 1=set -x
+debug_mode=0
+
+#############################################################################
+# parse the command line arguments
+
+parse_args() 
+{
+    while [[ "$#" -gt 0 ]]
+    do
+        arg_value="${2}"
+        shift_once=0
+
+        if [[ "${arg_value}" =~ "--" ]]; 
+        then
+            arg_value=""
+            shift_once=1
+        fi
+
+         # Log input parameters to facilitate troubleshooting
+        log "Option '${1}' set with value '"${arg_value}"'"
+
+        case "$1" in
+          --oxatools-public-github-accountname)
+            oxa_tools_public_github_account="${arg_value}"
+            ;;
+          --oxatools-public-github-projectname)
+            oxa_tools_public_github_projectname="${arg_value}"
+            ;;
+          --oxatools-public-github-projectbranch)
+            oxa_tools_public_github_projectbranch="${arg_value}"
+            ;;
+          --oxatools-public-github-branchtag)
+            oxa_tools_public_github_branchtag="${arg_value}"
+            ;;
+          --oxatools-repository-path)
+            oxa_tools_repository_path="${arg_value}"
+            ;;
+          --cluster-admin-email)
+            cluster_admin_email="${arg_value}"
+            ;;
+          --smtp-server)
+            smtp_server="${arg_value}"
+            ;;
+          --smtp-server-port)
+            smtp_server_port="${arg_value}"
+            ;;
+          --smtp-auth-user)
+            smtp_auth_user="${arg_value}"
+            ;;
+          --smtp-auth-user-password)
+            smtp_auth_user_password="${arg_value}"
+            ;;
+          --backend-server-list)
+            backend_server_list=(`echo ${arg_value} | base64 --decode`)
+            ;;
+          --target-user)
+            target_user="${arg_value}"
+            ;;
+          --remote)
+            remote_mode=1
+            ;;
+          --debug)
+            debug_mode=1
+            ;;
+        esac
+
+        shift # past argument or value
+
+        if [ $shift_once -eq 0 ]; 
+        then
+            shift # past argument or value
+        fi
+
+    done
+}
+
+validate_args()
+{
+    log "Validating arguments"
+
+    # target user
+    if [[ -z $target_user ]]; 
+    then
+        log "You must specify a user account to use for SSH to remote servers"
+        exit $ERROR_TOOLS_INSTALLER_FAILED
+    fi
+
+    # SMTP validation
+    if [[ -z $smtp_server ]] || [[ -z smtp_server_port ]] || [[ -z $smtp_auth_user ]] || [[ -z smtp_auth_user_password ]] || [[ -z cluster_admin_email ]];
```

Comment:
nit: we should stop passing secrets via commandline parameters as it will showup on disk if the console is getting logged and bash history if this script is being executed manually